src/swap/SwapScreen.test.tsx

```diff
@@ -1488,4 +1489,99 @@ describe('SwapScreen', () => {
     expect(queryByTestId('QuoteResultNotEnoughBalanceForGasBottomSheet')).toBeFalsy()
     expect(queryByTestId('QuoteResultNeedDecreaseSwapAmountForGasBottomSheet')).toBeFalsy()
   })
+
+  it("should display 'Fund your wallet' bottom sheet when user has no balance at all", async () => {
+    mockFetch.mockResponse(defaultQuoteResponse)
+
+    const store = createMockStore({
+      tokens: {
+        tokenBalances: {
+          [mockCusdTokenId]: {
+            ...mockTokenBalances[mockCusdTokenId],
+            isSwappable: true,
+            balance: '0',
+            priceUsd: '1',
+          },
+          [mockCeloTokenId]: {
+            ...mockTokenBalances[mockCeloTokenId],
+            isSwappable: true,
+            priceUsd: '1',
+            balance: '0',
+          },
+        },
+      },
+    })
+
+    const { getByText, getByTestId, getAllByTestId } = render(
+      <Provider store={store}>
+        <MockedNavigator component={SwapScreen} />
+      </Provider>
+    )
+    const [swapFromContainer, swapToContainer] = getAllByTestId('SwapAmountInput')
+    const tokenBottomSheet = getByTestId('TokenBottomSheet')
+
+    selectToken(swapFromContainer, 'CELO', tokenBottomSheet)
+    selectToken(swapToContainer, 'cUSD', tokenBottomSheet)
+    fireEvent.changeText(within(swapFromContainer).getByTestId('SwapAmountInput/Input'), '1')
+
+    await act(() => {
+      jest.runOnlyPendingTimers()
+    })
+
+    expect(getByText('swapScreen.confirmSwap')).not.toBeDisabled()
+    fireEvent.press(getByText('swapScreen.confirmSwap'))
+
+    expect(getByTestId('FundYourWalletBottomSheet')).toBeDefined()
```

Comment:
if i remember correctly, the bottom sheet is always defined in the tests (not only when it is snapped to the 0 index). perhaps it'd be worthwhile to test that swapStart is not dispatched?

Reply:
oh, good catch! i totally overlooked this. thanks for the suggestion!